bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsNodeSyncImpl.java

```diff
@@ -0,0 +1,609 @@
+                inputStream = this.dynamicClassLoaderManager.getDynamicClassLoader().getResourceAsStream("/remoteassets/remote_asset.zip");
+            } else {
+                inputStream = this.dynamicClassLoaderManager.getDynamicClassLoader().getResourceAsStream("/remoteassets/remote_asset.jpeg");
+            }
+
+            node.setProperty(JcrConstants.JCR_DATA, this.valueFactory.createBinary(inputStream));
+        } finally {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Binary added for node '{}'.", node.getPath());
+            }
+            inputStream.close();
+        }
+    }
+
+    /**
+     * Get the correct temporary binary (file type) based on file extension.
+     * @param parentNode Node
+     * @param file1 String
+     * @param file2 String
+     * @return InputStream
+     * @throws RepositoryException exception
+     */
+    private InputStream getCorrectBinaryTypeStream(final Node parentNode, final String file1, final String file2) throws RepositoryException {
+        Node assetNode = parentNode;
+        while (!DamConstants.NT_DAM_ASSET.equals(assetNode.getPrimaryNodeType().getName())) {
+            assetNode = assetNode.getParent();
+        }
+
+        String fileType = file1.equalsIgnoreCase(FilenameUtils.getExtension(assetNode.getName())) ? file1 : file2;
+        return this.dynamicClassLoaderManager.getDynamicClassLoader().getResourceAsStream("/remoteassets/remote_asset.".concat(fileType));
+    }
+
+    /**
+     * Set generic property for the property's node.
+     * @param value Object
+     * @param key String
+     * @param node Node
+     * @throws RepositoryException exception
+     */
+    private void setProperty(final Object value, final String key, final Node node) throws RepositoryException {
+        if (value instanceof String && DATE_REGEX.matcher((String) value).matches()) {
+            try {
+                node.setProperty(key, getFormattedDate((String) value));
+            } catch (DateTimeParseException e) {
+                LOG.warn("Unable to parse date '{}' for property:node '{}'.", value, key.concat(":").concat(node.getPath()));
+            }
+        } else if (value instanceof String && DECIMAL_REGEX.matcher((String) value).matches()) {
+            node.setProperty(key, new BigDecimal((String) value));
+        } else if (value instanceof Boolean) {
+            node.setProperty(key, (Boolean) value);
+        } else if (value instanceof Double) {
+            node.setProperty(key, (Double) value);
+        } else if (value instanceof Long) {
+            node.setProperty(key, (Long) value);
+        } else if (value instanceof Integer) {
+            node.setProperty(key, (Integer) value);
+        } else {
+            node.setProperty(key, value == null ? null : value.toString());
+        }
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Property '{}' added for node '{}'.", key, node.getPath());
```

Comment:
I wonder if setting each individual `property` (with maybe the exception of complex things like mixins) should be at the `trace` level.